pkgs/applications/window-managers/sawfish/default.nix

```diff
@@ -26,13 +27,27 @@ stdenv.mkDerivation rec {
     sha256 = "18p8srqqj9vjffg13qhspfz2gr1h4vfs10qzlv89g76r289iam31";
   };
 
-  nativeBuildInputs = [ autoreconfHook pkg-config ];
-  buildInputs = [ which
-    rep-gtk pango gdk-pixbuf-xlib imlib gettext texinfo
-    libXinerama libXrandr libXtst libICE libSM
-    makeWrapper ];
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+    gettext
+    makeWrapper
+    texinfo
+    which
+  ];
+  buildInputs = [
+    gdk-pixbuf-xlib
+    imlib
+    libICE
+    libSM
+    libXinerama
+    libXrandr
+    libXtst
+    pango
+    rep-gtk
+  ];
```

Comment:
```suggestion
  ];
  
  strictDeps = true;
```

Reply:
Why?

Reply:
To make sure that native and build Inputs are not mixed up again.

Reply:
When I use it, the configurations fails by not finding `rep`.

Reply:
That means that rep is missing in either nativeBuildInputs or buildInputs and won't work when cross compiling.

Reply:
But `rep` is a program installed in `librep`.

Reply:
za log:

```
warning: Git tree '/mnt/sources/WorkTrees/nixpkgs/updates' is dirty
sawfish> unpacking sources
sawfish> unpacking source archive /nix/store/fyzd66ibqwzb3sk74k6h447dyphk2qlm-sawfish_1.12.90.tar.xz
sawfish> source root is sawfish_1.12.90
sawfish> setting SOURCE_DATE_EPOCH to timestamp 1503693954 of file sawfish_1.12.90/sawfish.spec
sawfish> patching sources
sawfish> autoreconfPhase
sawfish> autoreconf: export WARNINGS=
sawfish> autoreconf: Entering directory '.'
sawfish> autoreconf: configure.ac: not using Gettext
sawfish> autoreconf: running: aclocal --force
sawfish> aclocal: warning: couldn't open directory 'm4': No such file or directory
sawfish> autoreconf: configure.ac: tracing
sawfish> autoreconf: running: libtoolize --copy --force
sawfish> libtoolize: putting auxiliary files in '.'.
sawfish> libtoolize: copying file './ltmain.sh'
sawfish> libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
sawfish> libtoolize: copying file 'm4/libtool.m4'
sawfish> libtoolize: copying file 'm4/ltoptions.m4'
sawfish> libtoolize: copying file 'm4/ltsugar.m4'
sawfish> libtoolize: copying file 'm4/ltversion.m4'
sawfish> libtoolize: copying file 'm4/lt~obsolete.m4'
sawfish> libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
sawfish> autoreconf: configure.ac: not using Intltool
sawfish> autoreconf: configure.ac: not using Gtkdoc
sawfish> autoreconf: running: aclocal --force
sawfish> autoreconf: running: /nix/store/y1f3d3b7jjcsix58vyksglagdmf2jyw0-autoconf-2.70/bin/autoconf --force
sawfish> configure.ac:24: warning: The macro `AC_CONFIG_HEADER' is obsolete.
sawfish> configure.ac:24: You should run autoupdate.
sawfish> ./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
sawfish> configure.ac:24: the top level
sawfish> configure.ac:27: warning: The macro `AM_PROG_LIBTOOL' is obsolete.
sawfish> configure.ac:27: You should run autoupdate.
sawfish> m4/libtool.m4:100: AM_PROG_LIBTOOL is expanded from...
sawfish> configure.ac:27: the top level
sawfish> configure.ac:60: warning: The macro `AC_HEADER_TIME' is obsolete.
sawfish> configure.ac:60: You should run autoupdate.
sawfish> ./lib/autoconf/headers.m4:743: AC_HEADER_TIME is expanded from...
sawfish> configure.ac:60: the top level
sawfish> autoreconf: running: /nix/store/y1f3d3b7jjcsix58vyksglagdmf2jyw0-autoconf-2.70/bin/autoheader --force
sawfish> autoreconf: configure.ac: not using Automake
sawfish> autoreconf: './config.sub' is updated
sawfish> autoreconf: './config.guess' is updated
sawfish> autoreconf: './install-sh' is updated
sawfish> autoreconf: Leaving directory '.'
sawfish> configuring
sawfish> fixing libtool script ./ltmain.sh
sawfish> configure flags: --disable-static --prefix=/nix/store/r1lcybw2c5ppm2p66n5cjnqxrb56zppb-sawfish-1.12.90
sawfish> checking build system type... x86_64-pc-linux-gnu
sawfish> checking host system type... x86_64-pc-linux-gnu
sawfish> checking how to print strings... printf
sawfish> checking for gcc... gcc
sawfish> checking whether the C compiler works... yes
sawfish> checking for C compiler default output file name... a.out
sawfish> checking for suffix of executables...
sawfish> checking whether we are cross compiling... no
sawfish> checking for suffix of object files... o
sawfish> checking whether the compiler supports GNU C... yes
sawfish> checking whether gcc accepts -g... yes
sawfish> checking for gcc option to enable C11 features... none needed
sawfish> checking for a sed that does not truncate output... /nix/store/n0516fv7vwjbl41nl4q58w9si80ab93i-gnused-4.8/bin/sed
sawfish> checking for grep that handles long lines and -e... /nix/store/3mx947xcmsyk79izqc8ifv31qafp80pc-gnugrep-3.6/bin/grep
sawfish> checking for egrep... /nix/store/3mx947xcmsyk79izqc8ifv31qafp80pc-gnugrep-3.6/bin/grep -E
sawfish> checking for fgrep... /nix/store/3mx947xcmsyk79izqc8ifv31qafp80pc-gnugrep-3.6/bin/grep -F
sawfish> checking for ld used by gcc... ld
sawfish> checking if the linker (ld) is GNU ld... yes
sawfish> checking for BSD- or MS-compatible name lister (nm)... nm
sawfish> checking the name lister (nm) interface... BSD nm
sawfish> checking whether ln -s works... yes
sawfish> checking the maximum length of command line arguments... 1572864
sawfish> checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
sawfish> checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
sawfish> checking for ld option to reload object files... -r
sawfish> checking for objdump... objdump
sawfish> checking how to recognize dependent libraries... pass_all
sawfish> checking for dlltool... no
sawfish> checking how to associate runtime and link libraries... printf %s\n
sawfish> checking for archiver @FILE support... @
sawfish> checking for strip... strip
sawfish> checking for ranlib... ranlib
sawfish> checking for gawk... gawk
sawfish> checking command to parse nm output from gcc object... ok
sawfish> checking for sysroot... no
sawfish> checking for a working dd... /nix/store/w21pgi6691202iafl0cnr7hk6wvvdz3n-coreutils-8.32/bin/dd
sawfish> checking how to truncate binary pipes... /nix/store/w21pgi6691202iafl0cnr7hk6wvvdz3n-coreutils-8.32/bin/dd bs=4096 count=1
sawfish> ./configure: line 6488: /usr/bin/file: No such file or directory
sawfish> checking for mt... no
sawfish> checking if : is a manifest tool... no
sawfish> checking for stdio.h... yes
sawfish> checking for stdlib.h... yes
sawfish> checking for string.h... yes
sawfish> checking for inttypes.h... yes
sawfish> checking for stdint.h... yes
sawfish> checking for strings.h... yes
sawfish> checking for sys/stat.h... yes
sawfish> checking for sys/types.h... yes
sawfish> checking for unistd.h... yes
sawfish> checking for sys/time.h... yes
sawfish> checking for dlfcn.h... yes
sawfish> checking for objdir... .libs
sawfish> checking if gcc supports -fno-rtti -fno-exceptions... no
sawfish> checking for gcc option to produce PIC... -fPIC -DPIC
sawfish> checking if gcc PIC flag -fPIC -DPIC works... yes
sawfish> checking if gcc static flag -static works... no
sawfish> checking if gcc supports -c -o file.o... yes
sawfish> checking if gcc supports -c -o file.o... (cached) yes
sawfish> checking whether the gcc linker (ld) supports shared libraries... yes
sawfish> checking whether -lc should be explicitly linked in... no
sawfish> checking dynamic linker characteristics... GNU/Linux ld.so
sawfish> checking how to hardcode library paths into programs... immediate
sawfish> checking whether stripping libraries is possible... yes
sawfish> checking if libtool supports shared libraries... yes
sawfish> checking whether to build shared libraries... yes
sawfish> checking whether to build static libraries... no
sawfish> checking for an ANSI C-conforming const... yes
sawfish> checking for inline... inline
sawfish> checking whether byte ordering is bigendian... no
sawfish> checking for size_t... yes
sawfish> checking for working alloca.h... yes
sawfish> checking for alloca... yes
sawfish> checking for vprintf... yes
sawfish> checking for gethostname... yes
sawfish> checking for socket... yes
sawfish> checking for snprintf... yes
sawfish> checking for getpagesize... yes
sawfish> checking for a BSD-compatible install... /nix/store/w21pgi6691202iafl0cnr7hk6wvvdz3n-coreutils-8.32/bin/install -c
sawfish> checking for xdr_void in -lnsl... no
sawfish> checking for bind in -lsocket... no
sawfish> checking for sys/wait.h that is POSIX.1 compatible... yes
sawfish> checking for fcntl.h... yes
sawfish> checking for sys/time.h... (cached) yes
sawfish> checking for sys/utsname.h... yes
sawfish> checking for pkg-config... /nix/store/1m90ppkamm4wk81l7ig65ircwiw7wiaa-pkg-config-wrapper-0.29.2/bin/pkg-config
sawfish> checking for X11/extensions/Xdbe.h... yes
sawfish> checking pkg-config is at least version 0.9.0... yes
sawfish> checking for xinerama >= 1.0
sawfish>                  ... yes
sawfish> checking for xrandr >= 1.0
sawfish>                  ... yes
sawfish> checking for xft >= 1.0
sawfish>                  ... yes
sawfish> checking for x11 >= 1.0
sawfish>                  ... yes
sawfish> checking for ice >= 1.0
sawfish>                  ... yes
sawfish> checking for sm >= 1.0
sawfish>                  ... yes
sawfish> checking for xtst >= 1.0
sawfish>                  ... yes
sawfish> checking for pango >= 1.8.0
sawfish>                  ... yes
sawfish> checking for pangoxft >= 1.8.0
sawfish>                  ... yes
sawfish> checking for rep... no
sawfish> rep binary not found!

```

Reply:
> But `rep` is a program installed in `librep`.

librep is only in buildInputs. It needs to go into nativeBuildInputs for the binaries to be in $PATH.